.github/workflows/rust.yml

```diff
@@ -8,8 +8,23 @@ defaults:
 
 jobs:
   static-analysis:
-    runs-on:
-      - ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        runs-on:
+          # We only need to run static analysis checks per OS family
+          - ubuntu-latest
+          - macos-13
+          - windows-2022
+        # TODO: https://github.com/rust-lang/cargo/issues/5220
+        include:
+          - runs-on: ubuntu-latest
+            packages: -p firezone-headless-client -p firezone-gateway -p connlib-client-android
```

Comment:
Can we just leave this empty? I run clippy on all rust packages all the time locally and it passes.

Reply:
I think in ubuntu yeah but in windows there are problems.
Being the case that we just can't compile everything in all platforms I just preferred being explicit in a per-platform basis, this way we are explicit on what package should run on what platform *always*

Reply:
Yeah I was specifically talking about the ubuntu case! I'd just like to have at least 1 CI job that lints all packages without having to manually list them. As we create new crates, this is likely to not get updated.

Reply:
Updated to lint all packages with a comment explaining why.